### PR TITLE
grpc: add version 1.66.1

### PR DESCRIPTION
**Pull request by itsmejoeeey**
### Summary
Changes to recipe:  ~~grpc/1.65.5~~ **grpc/1.66.1**

#### Motivation
The current package version has [a bunch of unwanted log output](https://github.com/grpc/grpc/issues/37178) that is fixed in the later minor releases.

This is the latest release.

#### Details
Minor fixes, see below:
https://github.com/grpc/grpc/compare/v1.65.0...v1.66.1

---
- [X] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by ErniGH** [comment]
Hey @itsmejoeeey , it looks like version 1.66.1 has been released. Do you still need version 1.65.5, or should we update the PR and add the latest one?

**Comment by itsmejoeeey** [comment]
Hi @ErniGH. For the purposes of my project, I'm not opposed to using the latest `grpc` version.

**Review by ErniGH — APPROVED** [review]
LGTM sorry for the delay ❤️

File: recipes/grpc/all/target_info/grpc_1.66.1.yml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        - utf8_range_lib
+        - upb_message_lib
+    - name: "upb_mem_lib"
+      lib: "upb_mem_lib"
+    - name: "upb_message_lib"
+      lib: "upb_message_lib"
+      requires:
+        - upb_base_lib
+        - upb_mem_lib
+    - name: "upb_textformat_lib"
```
Comment (AbrilRBS) [inline recipes/grpc/all/target_info/grpc_1.66.1.yml:96]: Seems like we might be missing https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L3847 `upb_mini_descriptor_lib`? This was added in 1.66

File: recipes/grpc/all/target_info/grpc_1.66.1.yml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      lib: "upb_textformat_lib"
+      requires:
```
Comment (AbrilRBS) [inline recipes/grpc/all/target_info/grpc_1.66.1.yml:98]: Same with https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L3983 `upb_wire_lib` Added in 1.66

File: recipes/grpc/all/target_info/grpc_1.66.1.yml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        - gpr
+        - address_sorting
+    - name: "grpc_plugin_support"
+      lib: "grpc_plugin_support"
+      requires:
+        - protobuf::libprotobuf
+        - protobuf::libprotoc
+    - name: "grpcpp_channelz"
+      lib: "grpcpp_channelz"
+      requires:
```
Comment (AbrilRBS) [inline recipes/grpc/all/target_info/grpc_1.66.1.yml:158]: https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L5847C13-L5847C31 `grpcpp_otel_plugin` missing too?

File: recipes/grpc/all/target_info/grpc_1.66.1.yml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    - name: "upb_message_lib"
+      lib: "upb_message_lib"
+      requires:
+        - upb_base_lib
+        - upb_mem_lib
+    - name: "upb_textformat_lib"
+      lib: "upb_textformat_lib"
+      requires:
+        - utf8_range_lib
+        - upb_message_lib
```
Comment (AbrilRBS) [inline recipes/grpc/all/target_info/grpc_1.66.1.yml:100]: This requirement is now outdated, https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L3713C3-L3713C15 these two are replaced by the new components noted above.

This is also the case for some other targets that now depend on the new libraries

**Review by AbrilRBS — COMMENTED** [review]
Thanks! @ErniGH and I have been looking into this and realized that the targets are outdated for this version (And perhaps some old one too, but that's another story)

We can help with the updating if needed, just let us know :)

**Comment by AbrilRBS** [comment]
Closing in favour of https://github.com/conan-io/conan-center-index/pull/25680, which is superseeding this PR while addressing my review comments, thanks a lot for taking the time to create the PR in the first time, we appreciate it even if the PR has not been merged :)

**Comment by itsmejoeeey** [comment]
Apologies for the delay in replying, got caught up in a hectic time at work.

Glad someone could carry on the torch :)

**Comment by AbrilRBS** [comment]
No worries, we appreciate the contribution regardless :)

**Pull request closed without merging**